web/vibe/web/common.d

```diff
@@ -479,6 +492,39 @@ package struct MethodAttribute
 	alias data this;
 }
 
+///
+public struct ResultSerializer (alias ST, alias DT, string ContentType)
+{
+	enum contentType = ContentType;
+	alias serialize = ST;
+	alias deserialize = DT;
+}
+
+alias resultSerializer(Args...) = ResultSerializer!(Args);
+
+///
+package alias DefaultSerializerT (FuncRetT) =
+	ResultSerializer!(
+		function (ref output_stream, ref value){
+			output_stream ~= serializeToJsonString(value);
```

Comment:
```suggestion
			serializeToJson(output_stream, value);
```

This can now use the range overload of `serializeToJson` to avoid the temporary copy. The parameter should be named `output_range`, tough, as this relies on a range interface rather than a stream one (same for `input_stream` below). It might make sense to go for a stream interface for certain serializers and large responses, but for now I think the range interface is more practical.

Reply:
thanks makes sense for the input, however I couldn't get rid of using streams for the output, please see:
https://github.com/vibe-d/vibe.d/pull/2492/files#r526105742

Reply:
> thanks makes sense for the output, however I couldn't get rid of using streams for the input, please see:
> https://github.com/vibe-d/vibe.d/pull/2492/files#r526105742
> 
> I could create an input range from the stream by copying it, but this would be inefficient

@s-ludwig for the follow up PR, are you okay with my suggestion? 
"I could create an input range from the stream by copying it, but this would be inefficient"